flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -106,28 +128,32 @@ public NetworkEnvironment(
 			new KvStateRegistry(),
 			null,
 			null,
+			new IOManagerAsync(),
 			IOManager.IOMode.SYNC,
 			partitionRequestInitialBackoff,
 			partitionRequestMaxBackoff,
 			networkBuffersPerChannel,
 			extraNetworkBuffersPerGate,
+			false,
 			enableCreditBased);
 	}
 
 	public NetworkEnvironment(
-		NetworkBufferPool networkBufferPool,
-		ConnectionManager connectionManager,
-		ResultPartitionManager resultPartitionManager,
-		TaskEventDispatcher taskEventDispatcher,
-		KvStateRegistry kvStateRegistry,
-		KvStateServer kvStateServer,
-		KvStateClientProxy kvStateClientProxy,
-		IOMode defaultIOMode,
-		int partitionRequestInitialBackoff,
-		int partitionRequestMaxBackoff,
-		int networkBuffersPerChannel,
-		int extraNetworkBuffersPerGate,
-		boolean enableCreditBased) {
+			NetworkBufferPool networkBufferPool,
+			ConnectionManager connectionManager,
+			ResultPartitionManager resultPartitionManager,
+			TaskEventDispatcher taskEventDispatcher,
+			KvStateRegistry kvStateRegistry,
+			KvStateServer kvStateServer,
+			KvStateClientProxy kvStateClientProxy,
+			IOManager ioManager,
+			IOMode defaultIOMode,
+			int partitionRequestInitialBackoff,
+			int partitionRequestMaxBackoff,
+			int networkBuffersPerChannel,
+			int extraNetworkBuffersPerGate,
+			boolean enableNetworkDetailedMetrics,
```

Comment:
Another option might we could pass `NetworkConfiguration` into the constructor instead of these specific parameters `partitionRequestInitialBackoff`, `partitionRequestMaxBackoff`, `networkBuffersPerChannel`, `extraNetworkBuffersPerGate`, `enableNetworkDetailedMetrics`, `enableCreditBased`. Then it seems better to not change this constructor always to add new parameters, just extend the `NetworkConfiguration`.

Reply:
I think it is good idea. At the end, `ShuffleManager.createShuffleService(config)` will probably need to accept Flink config to allow users set specific options for certain shuffle service.